src/erlcloud_route53.erl

```diff
@@ -0,0 +1,269 @@
+%% Amazon Route 53
+
+-module(erlcloud_route53).
+
+-export([describe_zones/0, describe_zones/3]).
+
+-export([describe_resource_sets/1, describe_resource_sets/4]).
+
+-export([describe_delegation_sets/0, describe_delegation_sets/3]).
+
+-include_lib("erlcloud/include/erlcloud.hrl").
+-include_lib("erlcloud/include/erlcloud_aws.hrl").
+-include_lib("erlcloud/include/erlcloud_route53.hrl").
+
+-define(API_VERSION, "2013-04-01").
+-define(DEFAULT_MAX_RECORDS, 20).
+-define(DESCRIBE_ZONES_PATH, "/ListHostedZonesResponse/HostedZones/HostedZone").
+-define(DESCRIBE_ZONE_IS_TRUNCATED, "/ListHostedZonesResponse/IsTruncated").
+-define(DESCRIBE_ZONE_NEXT_MARKER, "/ListHostedZonesResponse/NextMarker").
+-define(DESCRIBE_RS_PATH,
+        "/ListResourceRecordSetsResponse/ResourceRecordSets/ResourceRecordSet").
+-define(DESCRIBE_RS_IS_TRUNCATED,
+        "/ListResourceRecordSetsResponse/IsTruncated").
+-define(DESCRIBE_RS_NEXT_TYPE,
+        "/ListResourceRecordSetsResponse/NextRecordType").
+-define(DESCRIBE_RS_NEXT_NAME,
+        "/ListResourceRecordSetsResponse/NextRecordName").
+-define(DESCRIBE_RS_NEXT_ID,
+        "/ListResourceRecordSetsResponse/NextRecordIdentifier").
+-define(DESCRIBE_DS_PATH,
+        "/ListReusableDelegationSetsResponse/DelegationSets/DelegationSet").
+-define(DESCRIBE_DS_IS_TRUNCATED,
+        "/ListReusableDelegationSetsResponse/IsTruncated").
+-define(DESCRIBE_DS_NEXT_MARKER,
+        "/ListReusableDelegationSetsResponse/NextMarker").
+
+%% --------------------------------------------------------------------
+%% @doc Describes all zones using default config
+%% @end
+%% --------------------------------------------------------------------
+-spec(describe_zones() ->
+             {{paged, string()}, list(aws_route53_zone())} |
+             {ok, list(aws_route53_zone())} |
+             {error, term()}).
+describe_zones() ->
+    describe_zones([], ?DEFAULT_MAX_RECORDS, erlcloud_aws:default_config()).
+
+%% --------------------------------------------------------------------
+%% @doc Describes all zones using passed config, limit + AWS options
+%% @end
+%% --------------------------------------------------------------------
+-spec(describe_zones(Options   :: list({string(), string()}),
+                     Limit     :: integer(),
+                     AwsConfig :: aws_config()) ->
+    {{paged, string()}, list(aws_route53_zone())} |
```

Comment:
I'd also consider this discussion: https://github.com/gleber/erlcloud/pull/202#discussion_r38967721

about using flatreverse/1 function for optimizatiokn